src/python/visclaw/frametools.py

```diff
@@ -311,14 +386,22 @@ def plot_frame(framesolns,plotdata,frameno=0,verbose=False):
     if verbose:
         print '    Done with plotframe for frame %i at time %g' % (frameno,t)
 
-    
+
     # print the figure(s) to file(s) if requested:
     if (plotdata.mode() != 'iplotclaw') & plotdata.printfigs:
         # iterate over all figures that are to be printed:
-        for figno in plotted_fignos:
-            printfig(frameno=frameno, figno=figno, \
-                    format=plotdata.print_format, plotdir=plotdata.plotdir,\
-                    verbose=verbose)
+        for figname in plotdata._fignames:
+            plotfigure = plotdata.plotfigure_dict[figname]
+            figno = plotfigure.figno
+            if figno in plotted_fignos:
+                if figno in kml_fignos:
+                    printfig(frameno=frameno, figno=figno, \
+                             format=plotdata.print_format, plotdir=plotdata.plotdir,\
+                             verbose=verbose,kml_fig=True,kml_dpi=plotfigure.kml_dpi)
```

Comment:
This relates to a point above, you should not need a separate `kml_dpi` if you rely on the `ClawPlotFigure` `kwargs` attribute, something like `plotfigre.kwargs.get('dpi', 200)` should use a default if needed or use the one in the `kwargs` dictionary.